src/transformStateWithClones.js

```diff
@@ -7,7 +7,48 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const allStates = [];
+
+  for (const action of actions) {
+    const indexOfLastState = allStates.length - 1;
+    const lastState =
+      indexOfLastState >= 0 ? allStates[indexOfLastState] : { ...state };
+    const currentState = { ...lastState };
+
+    switch (true) {
+      case action.type === 'addProperties':
+        for (const key in action.extraData) {
+          if (Object.hasOwn(currentState, key)) {
```

Comment:
The method `Object.hasOwn` is not a standard JavaScript method. You might want to use `Object.prototype.hasOwnProperty.call(currentState, key)` instead to check if the property exists on the object.